### PR TITLE
[FIX] Bundling: merge dependency analysis results with raw module infos

### PR DESCRIPTION
**Pull request by codeworrior**
Fixes https://github.com/SAP/ui5-builder/issues/338

### CODE REVIEW

**Comment by codeworrior** [comment]
Note reg. the 2nd commit 265e5b64ea521993628e311ca3f27b6985692189: 
Due to the wrong handling of `findResource's` result, all bundles have been rated as 'decomposable' and the then needed but missing embedded resources caused 'uncaught promise rejection' errors.

Note to self: According to its name, `findResource` should not fail with an error but return null. Another `getResource` method could be introduced with the current behavior of `findResource` - if needed.

File: lib/lbt/bundle/Resolver.js [hunk 1]
```diff
@@ -56,8 +56,16 @@ class BundleResolver {
...
-				return moduleInfo && moduleInfo.subModules.length > 0 && !/(?:^|\/)library.js$/.test(moduleInfo.name);
+			function isDecomposableBundle(moduleInfo) {
+				if ( moduleInfo == null
+						|| moduleInfo.subModules.length === 0
+						|| /(?:^|\/)library.js$/.test(moduleInfo.name) ) {
+					return false;
+				}
+				// it might look more natural to expect 'all' embedded modules to exist in the pool,
+				// but expecting only one module is a more conservative approach
+				return Promise.all(moduleInfo.subModules.map((sub) => pool.findResource(sub).catch(() => false)))
```
Comment (matz3) [inline lib/lbt/bundle/Resolver.js]: As already [mentioned by @kristian](https://github.com/SAP/ui5-builder/commit/265e5b64ea521993628e311ca3f27b6985692189#r35226030) I think this won't work. The line is also not covered by any test.

Comment (codeworrior) [reply lib/lbt/bundle/Resolver.js]: Yes, you're both totally right.

**Comment by codeworrior** [comment]
Note: this PR requires a merge of PR https://github.com/SAP/ui5-builder/pull/341 to make the unit test for bundles work. Without that PR, the content of bundle modules is not properly recognized.

**Comment by codeworrior** [comment]
@kristian could you please test this PR again in your context? The new unit and integration tests should cover your scenario, too. But I might have missed something.

**Pull request merged**